### DISCUSSION TITLE
Parsing json output

### DISCUSSION THREAD

**Discussion by fitzyjoe** [rusq/slackdump#49] [Q&A] [answered]
Thanks @rusq.  Dumping worked very well.  Do you have a recommendation for parsing the json output?  I am using "jq" but it's obviously less useful than working in a UI like Slack.  Thx.

**Comment by rusq** [comment, marked as answer]
Hey @fitzyjoe , thanks for your donation!

`jq` is a great tool, but sometimes tedious to use.  I think the easiest option would be to use python, I added an example to the repository:
https://github.com/rusq/slackdump/blob/master/examples/python/print_messages.py

it requires that sample.json is present - which is a message dump from the slack (it's also in samples directory, if you need one)

**Reply by fitzyjoe** [reply]
Thanks, @rusq.  I'll have a look at python.  I saw your pull request.  I'll contribute a more complicated example to the `jq` example that you wrote.

**Reply by rusq** [reply]
Thank you for the contribution!  I will merge shortly :)

**Comment by rusq** [comment]
With `jq`, solution would be something like that (i haven't studied the man page enough to figure out how to get the thread replies conditionally)
```sh
#!/bin/sh

SAMPLE=../sample.json

cat "${SAMPLE}" | jq '.messages[]|.user +": "+.text' | tr -d '"'
```

**Comment by jingxuanlim** [comment]
Hi guys! Obviously very great work @rusq for building this utility. This was very helpful for me.

I thought I ask my question here instead of starting a new thread. Let me know if I should do otherwise.

I also had a question about parsing the json output and the reference to the example code helped me greatly.

Is there a way of automatically downloading attached files (such as images, videos and text files) so that they also display nicely?

I also wonder if there is an option, since attachments are downloaded into a folder, to have links to those files instead of to the original html link?
